### DISCUSSION TITLE
Midi Actions

### DISCUSSION THREAD

**Discussion by DenishStrat** [ChurchApps/ChurchAppsSupport#319] [FreeShow] [answered]
Hi, first I wanna thank you for that wonderfull app! 
I use ProPresenter on a macbook automated by Midi with Logic Pro X (all on the same computer). It always work great.
I tried with FreeShow but I cannot communicate with Logic... I try to trigger the next slide action sending Midi from Logic but nothing happen... I created an Input for FreeShow (bus) and nothing... Can you tell me what to do or make a small tutorial on what to do on Mac to trigger Midi Action from a Daw like Logic for exemple?

Thanks!

**Comment by vassbo** [comment]
Hi, did you go into settings -> "Actions" -> "Add"? From the popup you have to select the midi input which should come from Logic.
After that you need to send with the correct note and channel in Logic.
See more here: https://freeshow.app/docs/midi

If you don't see any midi inputs please let me know, and I'll investigate.

**Reply by DenishStrat** [reply]
Yes that's what I did... I'm doing the same kind of automation with ProPresenter and everything works..

I have the right Input. I created an Action (Play On Midi In) for the first slide of a song (show), just to select that song.
I also created a Next Slide Action in the Setting \ Actions but nothing is happening when Logic send those midi data... 

In ProPresenter there is an option to enable the connection (communication) with the Midi Bus (Input)...
In FreeShow I only have to select the right Input for every Actions?

**Comment by vassbo** [comment]
With the popup open, disable the "Use default values" option. Then enable "Update values when receiving a MIDI input", and check if the values changes when receiving a note.

**Reply by DenishStrat** [reply]
In the Setting \ Action tab, if I edit an Action, when I enable "Update values when receiving a MIDI input" then quit the popup window by clicking the X it's like if nothing was saved... If I return in the popup window by clicking the little pencil (to edit the action) the option "Update values when receiving a MIDI input" is still disabled.... If I enble it again it will be disabled the next time I open it..... It's like if the "enable button" is doing nothing.... You understand?

**Comment by vassbo** [comment, marked as answer]
The "Update values when receiving a MIDI input" is just for setting the tone/values automatically, it will be disabled again when closing the popup, because you want the same note to trigger the same action! Did you try sending a MIDI note from Logic with the popup open, and did the values change when the slider was toggled?

**Comment by Ruk94** [comment]
Hi [vassbo](https://github.com/vassbo)!
Absolute newbie here, wondering how flexible is MIDI. 
Can I assign any midi note to any slide in FreeShow? Meaning, can I assign say, 20 keys on my keyboard to play 20 specific slides? 
Is there any delay? 
I would like to use my DAW to trigger images and slides in sync with the music. 
Thank you!

**Comment by vassbo** [comment]
Hi, this should be possible.

You should be able to set any specific MIDI signals per slide (right click, "Slide actions", "Activate on MIDI signal").
But I think it's easier to set a custom MIDI action per slide "group" instead, or using "Next slide" action ("Functions">"Actions").
The delay should not be noticeable.
